client/my-sites/domains/domain-management/dns/dns-add-new.jsx

```diff
@@ -180,34 +225,32 @@ class DnsAddNew extends React.Component {
 		return ! formState.isFieldInvalid( this.state.fields, fieldName );
 	};
 
-	recordFields() {
-		return this.dnsRecords.map( ( dnsRecord ) => {
-			const { component: Component, types: showTypes } = dnsRecord;
-
-			return (
-				<Component
-					key={ showTypes.join( ',' ) }
-					selectedDomainName={ this.props.selectedDomainName }
-					show={ includes( showTypes, this.state.fields.type.value ) }
-					fieldValues={ formState.getAllFieldValues( this.state.fields ) }
-					isValid={ this.isValid }
-					onChange={ this.onChange }
-				/>
-			);
-		} );
+	renderFields( selectedRecordType ) {
+		return (
+			<selectedRecordType.component
+				selectedDomainName={ this.props.selectedDomainName }
+				show={ true }
```

Comment:
I believe this property can now be removed from the record components, let's do that in a later PR.